ddtrace/tracer/tracer.go

```diff
@@ -351,16 +372,29 @@ func (t *tracer) Extract(carrier interface{}) (ddtrace.SpanContext, error) {
 }
 
 // flush will push any currently buffered traces to the server.
-func (t *tracer) flushPayload() {
+func (t *tracer) flushPayload(confirm chan<- struct{}) {
 	if t.payload.itemCount() == 0 {
+		if confirm != nil {
+			confirm <- struct{}{}
+		}
 		return
 	}
+	pl := t.payload
+	t.payload = t.payloads.Get().(*payload)
```

Comment:
Generally the items in the pool are reset after being "Get", not before being "Put" ([example](https://github.com/golang/go/blob/50bd1c4d4eb4fac8ddeb5f063c099daccfb71b26/src/archive/zip/register.go#L70)).

Reply:
The only explanation I can think of is that it might be a waste of time to reset something that is being put back into the pool, knowing that the object may be removed from the pool without notice, never to be used again (https://golang.org/pkg/sync/#Pool)